### ISSUE TITLE
Defined constants are dropped if using environment specific configurations

### ISSUE THREAD

**Issue by danwkennedy** [ajwhite/gulp-ng-config#21]
I'm using the constants option to inject build info into Angular (latest commit SHA, branch, tag...) and I'm also using environment dependent configurations.

config.json 
```
{
     "prod": {},
     "dev": {}
}
```
And in gulpFile.js:
```
   gulp
       .src(appPath + '/config.json')
       .pipe(gulpNgConfig('app.config', {
            environment: args.environment,
            constants: {
                environment: args.environment,
                buildInfo: buildInfo
            }
        }))
        .pipe(gulp.dest(appPath));
```
Right now, the environment dependent configs win over constants and the generated config omits "environment" and "buildInfo". I've figured out what the issue was and it was a very simple fix. Creating a PR for it.

**Comment by ajwhite** [comment]
Thanks!

**Comment by danwkennedy** [comment]
It's in and unit tested

**Comment by ajwhite** [comment]
This has been published as part of `v1.2.0`
